### PR TITLE
fix(i): Null compound filter panic

### LINKED ISSUE #3079: Null compound filter causes panic
Null compound filters and most likely null elements cause a panic.

Example: 

```json
{
  "filter": {
    "_or": null
  }
}
```

Results:

```
panic: interface conversion: interface {} is nil, not []interface {}
 
 -> github.com/sourcenetwork/defradb/internal/planner/mapper.resolveInnerFilterDependencies
 ->   github.com/sourcenetwork/defradb/internal/planner/mapper/mapper.go:969
```

### PR DESCRIPTION
**Pull request by nasdf**
## Relevant issue(s)

Resolves #3079

## Description

This PR fixes an issue where null values within compound filters would panic.

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

Added integration tests

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

File: internal/planner/mapper/mapper.go [hunk 1]
```diff
@@ -966,7 +966,10 @@ func resolveInnerFilterDependencies(
 
 	for key := range source {
 		if key == request.FilterOpAnd || key == request.FilterOpOr {
-			compoundFilter := source[key].([]any)
+			compoundFilter, ok := source[key].([]any)
+			if !ok {
+				continue // filter is nil
```
Comment (AndrewSisley) [inline internal/planner/mapper/mapper.go:971]: suggestion: I do think `if foo == nil` conveys intent much better than a cast+comment

**Review by AndrewSisley — APPROVED** [review]
LGTM, but I really think you should be checking for nil with an 'if nil' check instead of a failed cast.  Might be worth raising with the team, as I've mentioned this before to you and I don't want to be pushing you away from it if it is only me who thinks this is a bad way of handling it.

File: internal/planner/mapper/mapper.go (hunk 1 above)
Comment (fredcarle) [reply internal/planner/mapper/mapper.go:971]: I don't mind the type assertion here as it does 2 things. It ensures it's an array and not nil at the same time.

Comment (AndrewSisley) [reply internal/planner/mapper/mapper.go:971]: I see that as part of the problem, it shouldn't be checking the type here, and it is not really handling it if it is somehow the wrong type - it is just silently ignoring what could be a serious coding error.

**Pull request merged**